### ISSUE TITLE
Division by zero crashes hardware, but not cen64.

### ISSUE THREAD

**Issue by joeldipops** [n64dev/cen64#126]
I'm writing a project in libdragon and at one point I tried calling 
```
frequency = 0;
buffers = 4;
audio_init(frequency, buffers);
```
This runs perfectly fine in cen64 but my PAL n64 with Everdrive 2.5 crashes immediately to a solid black screen.

After a quick investigation I found that the first mention of `frequency` in `audio_init` is on line 30
`AI_regs->dacrate = ((2 * clockrate / frequency) + 1) / 2 - 1;`
So it looks like a classic divide by 0 that cen64 doesn't replicate the behaviour of.

https://github.com/DragonMinded/libdragon/blob/122a61546bac17f36537b1b73d2ea8758e188f3f/src/audio.c

**Comment by sp1187** [comment]
This is most likely because the trap instructions are not yet implemented in CEN64. In your case it would be that the code run a trap on equal instruction comparing the divisor register with zero.

**Comment by tj90241** [comment]
yup - need to verify what the "undefined behavior" is too.

```
The contents of general purpose register rs are divided by the contents of general
purpose register rt, treating both operands as unsigned integers. An integer
overflow exception never occurs, and the result of this operation is undefined
when the divisor is zero. In 64-bit mode, the result must be sign-extended, 32-bit
values.
This instruction is executed after the instructions to check for a zero division.
```

**Comment by sp1187** [comment]
The integer division of zero division behavior has already been tested and was implemented in https://github.com/n64dev/cen64/commit/65b5a08cd612dcfead72cfb12f7b405df74d7061. It's also described in page 631 of http://n64dev.org/p/U10504EJ7V0UMJ1.pdf (seems like the VR4400 is the one that is actually used).

**Comment by joeldipops** [comment]
Got latest cen64 master and rebuilt tonight and behaviour is the same as before.  Unless my 64 is faulty or...?

**Comment by sp1187** [comment]
Yes, because the trap instructions, which is generated by the compiler and actually performs the zero sanity check, has still not been implemented. The div instruction itself just produces a dummy value, and that dummy value is generated properly, that's not the issue here.

**Comment by sp1187** [comment]
Closing as this should have been fixed by #159 (if using traps) or #150 (if using breaks).
